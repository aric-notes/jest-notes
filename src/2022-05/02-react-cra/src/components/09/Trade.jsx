import React from 'react';

const Trade = ({ wood }) => {
  function sellWood(wood) {
    const min = wood * 100;
    const max = wood * 500;
    return Math.floor(Math.random() * (max - min + 1)) + min;
  }

  return (
    <div>
      <h3>{wood}個木頭拿去交易</h3>
      <div data-testid='sell'>賣完，得到 ${sellWood(wood)}</div>
    </div>
  );
};

export default Trade;