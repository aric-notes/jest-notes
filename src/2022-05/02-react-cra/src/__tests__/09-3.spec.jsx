import { render } from '@testing-library/react';
import Trade from '../components/09/Trade';


test('<Trade />', () => {
  // Arrange
  // randomMock = jest.spyOn(global.Math, "random");
  jest.spyOn(global.Math, 'random').mockReturnValue(0.1);

  const { container } = render(<Trade wood={50} />);

  expect(container).toMatchSnapshot()

});