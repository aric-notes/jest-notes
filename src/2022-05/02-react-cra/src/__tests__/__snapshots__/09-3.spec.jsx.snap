// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Trade /> 1`] = `
<div>
  <div>
    <h3>
      50
      個木頭拿去交易
    </h3>
    <div
      data-testid="sell"
    >
      賣完，得到 $
      7000
    </div>
  </div>
</div>
`;
